Speed up unit tests compilation

Novakov opened the pull request:
According to https://github.com/google/googletest/blob/master/googlemock/docs/CookBook.md#making-the-compilation-faster it is wise to define constructor and destructor of mock class in single .cpp file to speed up compilation.

On my machine, command: `make -C unit_tests clean all -j5`
Before: ~67s
After: ~57s

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pw-sat2/pwsat2obc/87)
<!-- Reviewable:end -->

zaq32 commented:
You can also gain something by putting definition of all of the newly added constructors/destructors in single file.

---

Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[unit_tests/OsMock.hpp, line 12 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/87#-KcWT1n6znjulm0Vzvx1:-KcWT1n6znjulm0Vzvx2:b-lo4mia) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/7c600bbd163e394ec7e6ef5644da1fc21ae1a65f/unit_tests/OsMock.hpp#L12)):*
> ```cpp
> {
>     OSMock();
>     virtual ~OSMock();
> ```

IOS does not have virtual destructor so why this one is marked as virtual?

---

*[unit_tests/I2C/I2CMock.hpp, line 12 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/87#-KcWTEOmbPrY8rYG8J_w:-KcWTEOmbPrY8rYG8J_x:b-4vom91) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/7c600bbd163e394ec7e6ef5644da1fc21ae1a65f/unit_tests/I2C/I2CMock.hpp#L12)):*
> ```cpp
> {
>     I2CBusMock();
>     virtual ~I2CBusMock();
> ```

drivers::i2c::II2CBus does not have virtual destructor so why this one is marked as virtual?

---

*[unit_tests/mock/FsMock.hpp, line 13 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/87#-KcWTICr96ksTjqjFH33:-KcWTICsk8tGcQVBDflx:b-4c4q8p) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/7c600bbd163e394ec7e6ef5644da1fc21ae1a65f/unit_tests/mock/FsMock.hpp#L13)):*
> ```cpp
> {
>     FsMock();
>     virtual ~FsMock();
> ```

IFileSystem does not have virtual destructor so why this one is marked as virtual?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/87#-:-KcWTM9OZH00LA2rsuuc:b-69y4yn)*
<!-- Sent from Reviewable.io -->

Novakov commented:
Well it's just two files that will remain mostly the same. Putting both ctors in single file will make a bit harder to find

---

Review status: 3 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[unit_tests/OsMock.hpp, line 12 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/87#-KcWT1n6znjulm0Vzvx1:-KcdLFnmq2VhsXYqErWU:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/7c600bbd163e394ec7e6ef5644da1fc21ae1a65f/unit_tests/OsMock.hpp#L12)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

IOS does not have virtual destructor so why this one is marked as virtual?
</blockquote></details>

Fixed

---

*[unit_tests/I2C/I2CMock.hpp, line 12 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/87#-KcWTEOmbPrY8rYG8J_w:-KcdLI_NOS-a350CWd-K:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/7c600bbd163e394ec7e6ef5644da1fc21ae1a65f/unit_tests/I2C/I2CMock.hpp#L12)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

drivers::i2c::II2CBus does not have virtual destructor so why this one is marked as virtual?
</blockquote></details>

Fixed

---

*[unit_tests/mock/FsMock.hpp, line 13 at r1](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/87#-KcWTICr96ksTjqjFH33:-KcdLJMYWsm-aQlw7Ukw:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/7c600bbd163e394ec7e6ef5644da1fc21ae1a65f/unit_tests/mock/FsMock.hpp#L13)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary><blockquote>

IFileSystem does not have virtual destructor so why this one is marked as virtual?
</blockquote></details>

Fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/87#-:-KcdIAlHu1KF1f2gdwm4:b-lbaqd9)*
<!-- Sent from Reviewable.io -->

zaq32 commented:
just a suggestion...

---

Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/87#-:-KcdUNQVxtsdrj408Jmo:b-ijgnk4)*
<!-- Sent from Reviewable.io -->
